clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1993,6 +1993,17 @@ device kernel, the attribute is ignored and it is not propagated to a kernel.
   }];
 }
 
+def SYCLIntelMaxGlobalWorkDimAttrDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "max_global_work_dim (IntelFPGA)";
+  let Content = [{
+Applies to a device function/lambda function. Indicates the largest valid global
```

Comment:
Reading the code below, this attribute can only be applied to a SYCL Kernel function, i.e. to `operator()` of a globally-visible type, which takes certain arguments. I think this restriction should be in the docs

Reply:
Added the info.

Reply:
`device function` part is still not fully correct: according to the [code](https://github.com/intel/llvm/pull/915/files#diff-c88a590aca6d5f8e92dd1f8b6ec05a49R480), you will drop the attribute if it is applied not to a SYCL Kernel function

Reply:
In a perfect world, we support right-side (aka function type attributes) without attribute propagation to the caller (including device functions). For now it's not a case, and specification is vague and fragile. I would leave the documentation as is.